bundles/framework/layerlist/view/LayerViewTabs/LayerList/LayerCollapse/util.js

```diff
@@ -45,26 +45,66 @@ export const groupLayers = (layers, method, tools, allGroups = [], allDataProvid
         return typeof groupId === 'number' ? groupId : -Math.random();
     };
 
+    const mapGroupsAndLayers = function (group) {
+        // group has subgroups
+        if (group.hasSubgroups()) {
+            // check that group has layers
+            if (group.layers && group.layers.length > 0) {
```

Comment:
Both if statements are wrapped in the same ` if (group.layers && group.layers.length > 0) {` making already complicated method more nested than is should be.